File: BackEnd/backend/src/main/java/com/fitapp/backend/routinecomplete/package_/aplication/dto/request/PackageFilterRequest.java
package com.fitapp.backend.routinecomplete.package_.aplication.dto.request;


import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Schema(description = "Filters for marketplace search")
public class PackageFilterRequest {
 
    @Schema(description = "Search in name/description")
    @JsonProperty("search")
    private String search;
 
    @Schema(description = "Filter by package type")
    @JsonProperty("packageType")
    private String packageType;
 
    @Schema(description = "Filter by free/paid")
    @JsonProperty("isFree")
    private Boolean isFree;
 
    @Schema(description = "Filter by minimum subscription requirement")
    @JsonProperty("requiresSubscription")
    private String requiresSubscription;
 
    @Schema(description = "Filter by rating (minimum)")
    @JsonProperty("minRating")
    private Double minRating;
 
    @Schema(description = "Filter by creator (user ID, null = official packs)")
    @JsonProperty("createdByUserId")
    private Long createdByUserId;

    @Builder.Default
    @Schema(description = "Sort field: createdAt, rating, downloadCount, name")
    @JsonProperty("sortBy")
    private String sortBy = "createdAt";
 
    @Builder.Default
    @Schema(description = "Sort direction: ASC, DESC")
    @JsonProperty("sortDirection")
    private String sortDirection = "DESC";
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/feature/marketplace/model/request/PackageFilterRequest.kt
package com.fitapp.appfit.feature.marketplace.model.request

import com.google.gson.annotations.SerializedName

data class PackageFilterRequest(
    @SerializedName("search")
    val search: String? = null,

    @SerializedName("packageType")
    val packageType: String? = null,

    @SerializedName("isFree")
    val isFree: Boolean? = null,

    @SerializedName("requiresSubscription")
    val requiresSubscription: String? = null,

    @SerializedName("minRating")
    val minRating: Double? = null,

    @SerializedName("createdByUserId")
    val createdByUserId: Long? = null,

    @SerializedName("sortBy")
    val sortBy: String = "createdAt",

    @SerializedName("sortDirection")
    val sortDirection: String = "DESC"
)